docs/sources/reference/commandline/cli.md

```diff
@@ -637,13 +637,23 @@ an [*ADD*](/reference/builder/#add) instruction to reference a file in the
 context.
 
 The `URL` parameter can specify the location of a Git repository;
-the repository acts as the build context.  The system recursively clones the repository
+the repository acts as the build context. The system recursively clones the repository
 and its submodules using a `git clone --depth 1 --recursive` command.
 This command runs in a temporary directory on your local host.
 After the command succeeds, the directory is sent to the Docker daemon as the context.
 Local clones give you the ability to access private repositories using
 local user credentials, VPN's, and so forth.
 
+Git urls accept context configuration in their fragment section divided by a colon `:`.
+The first part represents the reference that git will check out, this can be either
+a branch, a tag or a commit sha. The second part represents a subdirectory
+inside the main root repository that will be used as a root context.
+
+For example, you need to run this command to use a directory called `docker`,
+in the branch `container`, as a context:
+
+  docker build https://github.com/docker/rootfs.git#container:docker
+
```

Comment:
We do support tables. You need to tick it up and do some tweaks to your headings.
@calavera ping me direct if you have specific questions on the markdown

--->

For example, as context, you need to run this command to use a directory called `docker` in the branch `container`:

  		$ docker build https://github.com/docker/rootfs.git#container:docker

The following table represents all the valid suffixes with their build contexts:

Build Syntax Suffix | Commit Used | Build Context Used
--------------------|-------------|-------------------
`myrepo.git` | `refs/heads/master` | `/`
`myrepo.git#mytag` | `refs/tags/mytag` | `/`
`myrepo.git#mybranch` | `refs/heads/mybranch` | `/`
`myrepo.git#abcdef` | `sha1 = abcdef` | `/`
`myrepo.git#:myfolder` | `refs/heads/master` | `/myfolder`
`myrepo.git#master:myfolder` | `refs/heads/master` | `/myfolder`
`myrepo.git#mytag:myfolder` | `refs/tags/mytag` | `/myfolder`
`myrepo.git#mybranch:myfolder` | `refs/heads/mybranch` | `/myfolder`
`myrepo.git#abcdef:myfolder` | `sha1 = abcdef` | `/myfolder`

Instead of specifying a context, you can pass a single Dockerfile in the
`URL` or pipe the file in via `STDIN`. To pipe a Dockerfile from `STDIN`:

	 $ docker build - < Dockerfile

If you use STDIN or specify a `URL`, the system places the contents into a
file called `Dockerfile`, and any `-f`, `--file` option is ignored. In this
scenario, there is no context.